### PR TITLE
Update deepcopy to work cross platforms and without GOPATH

### PR DESCRIPTION
**Pull request by JoelSpeed**
As a follow up to #1257 and in an effort to try and remove our tooling dependencies from the root go module, this PR updates the DeepCopy script to work independent of GOPATH settings and also starts to remove the code-generator dependencies from the root module.

I noticed an out of date generation as well so have updated that.

I will follow this up with a PR to update the protobuf generation which will then mean we can remove all of the code-generator dependency from the root module.

I've tested this on Mac and Linux (container), inside and outside of the GOPATH on my system

CC @mdbooth @Miciah

### CODE REVIEW

**Comment by deads2k** [comment]
/approve
/hold

holding so a bash knowledgeable person can lgtm.

File: hack/update-deepcopy.sh [hunk 1]
```diff
@@ -2,17 +2,28 @@
...
-  "${inputArg}" \
-  --go-header-file ${SCRIPT_ROOT}/hack/empty.txt \
+# API_GROUP_VERSION_PACKAGES is a string of <group>/<version>.
+# The deepcopy gen needs a comma-separated list of ./<group>/<version> so print in that format and remove the leading comma.
+inputArg="$(printf ",./%s" ${API_GROUP_VERSIONS})"
+inputArg="${inputArg:1}"
+
+echo Generating Deepcopy for ${API_GROUP_VERSIONS}
+
+GOPATH= ${DEEPCOPY_GEN} \
```
Comment (mdbooth) [inline hack/update-deepcopy.sh:24]: nit: You should quote `"${DEEPCOPY_GEN}"` here.

In practise it's only going to break anybody with whitespace in their `${SCRIPT_ROOT}`, so possibly nobody, but it's best practise.

Comment (mdbooth) [inline hack/update-deepcopy.sh:24]: An alternative to the leading `GOPATH=` here would be to `unset GOPATH` up top with a comment.

**Review by mdbooth — COMMENTED** [review]
I tested this in my environment where I don't use GOPATH and GOPATH is not set. I ran:

```bash
# Delete all generated deepcopy files
find -name "zz_*.deepcopy.go" -delete

# Restore those from vendor directories
git checkout vendor tools/vendor

# Regenerate package deepcopy files
make update-deepcopy
```

There was no resulting diff, so this is working correctly 😁

Couple of nits if you want to address them.
/lgtm

File: hack/update-deepcopy.sh (hunk 1 above)
Comment (JoelSpeed) [reply hack/update-deepcopy.sh]: I wanted to have this here so it's very obvious that we are clearing it for this command execution, added a comment

**Comment by mdbooth** [comment]
Thanks for addressing the nits! I also retested and this is still good in my environment.

/lgtm

**Comment by JoelSpeed** [comment]
/hold cancel

Thanks for the review @mdbooth

**Pull request merged**